src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -86,12 +88,14 @@ internal class LocalGrainDirectory :
         internal readonly CounterStatistic UnregistrationsManyRemoteSent;
         internal readonly CounterStatistic UnregistrationsManyRemoteReceived;
 
-        
         public LocalGrainDirectory(Silo silo)
         {
             log = LogManager.GetLogger("Orleans.GrainDirectory.LocalGrainDirectory");
 
             MyAddress = silo.LocalMessageCenter.MyAddress;
+            isSiloNextInTheRing = (siloAddr, hash, excludeMySelf) => (siloAddr.GetConsistentHashCode() <= hash) &&
+                                                                     (!excludeMySelf || !siloAddr.Equals(MyAddress));
```

Comment:
@jdom I agree with the first point.
For the second, when you pass an instance method as a delegate, like
```C#
membershipRingList.FindLast(this.SomePredicateMethod)
```
you are allocating a new `Predicate<T>` every time.